mpd/mpd.go

```diff
@@ -218,19 +263,31 @@ type AudioChannelConfiguration struct {
 
 // Creates a new MPD object.
 // profile - DASH Profile (Live or OnDemand).
-// mediaPresentationDuration - Media Presentation Duration (i.e. PT6M16S).
 // minBufferTime - Min Buffer Time (i.e. PT1.97S).
-func NewMPD(profile DashProfile, mediaPresentationDuration string, minBufferTime string) *MPD {
+// attributes - Other attributes (optional)
+func NewMPD(profile DashProfile, minBufferTime string, attributes ...AttrMPD) *MPD {
 	period := &Period{}
-	return &MPD{
-		XMLNs:    Strptr("urn:mpeg:dash:schema:mpd:2011"),
-		Profiles: Strptr((string)(profile)),
-		Type:     Strptr("static"),
-		MediaPresentationDuration: Strptr(mediaPresentationDuration),
-		MinBufferTime:             Strptr(minBufferTime),
-		period:                    period,
-		Periods:                   []*Period{period},
+	mpd := &MPD{
+		XMLNs:         Strptr("urn:mpeg:dash:schema:mpd:2011"),
+		Profiles:      Strptr((string)(profile)),
+		Type:          Strptr("static"),
+		MinBufferTime: Strptr(minBufferTime),
+		period:        period,
+		Periods:       []*Period{period},
+	}
+
+	for i := range attributes {
+		switch attr := attributes[i].(type) {
+		case *attrAvailabilityStartTime:
+			mpd.AvailabilityStartTime = attr.GetStrptr()
+		case *attrMinimumUpdatePeriod:
+			mpd.MinimumUpdatePeriod = attr.GetStrptr()
+		case *attrMediaPresentationDuration:
+			mpd.MediaPresentationDuration = attr.GetStrptr()
+		}
```

Comment:
Nice way to add attributes, should add some validation too and along with that some negative test cases

Reply:
Is the validation you are talking about attributes that do not match type, or about the format of attribute values?